Bpa configclasses

BennettAllen1 opened the pull request:
Hi,
here are the plugins, these are called in the .install in the debians where the config_files then apply to the node eg. dns_json and shared_config are in clearwater-etcd/debian/clearwater-config-manager.install
and then the rest are spread across sprout, Houdini, chronos and weatherwax in debian/*.install

BennettAllen1 committed:
This adds config type plugins which are used for config changes

There are 10 sub classes each describing one config type and 1 main class which they all use

BennettAllen1 committed:
Updates config plugins to call validation correctly

this updates the way to call the scripts to validate config

BennettAllen1 committed:
This adjusts the validation function

this edits the sub class and main class to allow for the validation scripts to be called to check the correct file path for the local copy

BennettAllen1 committed:
Edits description of what needs to be done

BennettAllen1 committed:
Adds a help info description to all the config_plugins

BennettAllen1 committed:
This adjusts the location to find the base class which all the sub classes call upon

BennettAllen1 committed:
changes a file name

BennettAllen1 committed:
adds file name change

BennettAllen1 committed:
Merge branch 'bpa_configclasses' of github.com:Metaswitch/clearwater-etcd-plugins into bpa_configclasses

Conflicts:
	clearwater_config_access/shared_config_config_plugin.py

BennettAllen1 committed:
adds a docstring

BennettAllen1 committed:
Merge branch 'bpa_configclasses' of github.com:Metaswitch/clearwater-etcd-plugins into bpa_configclasses

Conflicts:
	clearwater_config_access/shared_config_config_plugin.py

BennettAllen1 committed:
Adds a file describer to the script

this allows the correct validation script finder to be used for xml and json files

BennettAllen1 committed:
Merge branch 'bpa_configclasses' of github.com:Metaswitch/clearwater-etcd-plugins into bpa_configclasses

BennettAllen1 committed:
Gives each subclass a filetype

BennettAllen1 committed:
Merge branch 'bpa_configclasses' of github.com:Metaswitch/clearwater-etcd-plugins into bpa_configclasses

BennettAllen1 committed:
This removes the scripts call in the class structure

BennettAllen1 committed:
Merge branch 'bpa_configclasses' of github.com:Metaswitch/clearwater-etcd-plugins into bpa_configclasses

BennettAllen1 committed:
Adds an __init__.py file to the config_access folder

This allows the config_access to be accessed as a module by python

BennettAllen1 committed:
Merge branch 'bpa_configclasses' of github.com:Metaswitch/clearwater-etcd-plugins into bpa_configclasses

BennettAllen1 committed:
Merge remote-tracking branch 'origin/master' into bpa_configclasses

BennettAllen1 committed:
Merge branch 'bpa_configclasses' of github.com:Metaswitch/clearwater-etcd-plugins into bpa_configclasses

BennettAllen1 committed:
Updates the validation scripts for shared config

BennettAllen1 committed:
Merge branch 'bpa_configclasses' of github.com:Metaswitch/clearwater-etcd-plugins into bpa_configclasses

RobDover reviewed (commented):
I *think* I can piece together what one of these is supposed to look like, but it's not at all obvious what each of the fields does. I think you need to make that a lot more clear in the base class so that there's a place to reference what they all do.

I think there's probably a lot of redundancy in all of the settings your configuring for each type - once you've written down the definitions of what each of these are in the base class I'd like to revisit these and go over whether we've got the right settings of whether they need changing.

BennettAllen1 committed:
Makes changes to all the subclasses of config type to add description

Adds description in the form of more comments and docstrings

eleanor-merry commented:
Can you make sure you add the standard copyright statement whenever you add a new file please?

BennettAllen1 committed:
Adds copywright statement and docstring to all files

BennettAllen1 committed:
Updates comment on help_info

BennettAllen1 committed:
Removes file

This file has moved into houdini/scripts

BennettAllen1 committed:
removes weatherwax from plugins repo

This has been moved to weatherwax repo

BennettAllen1 committed:
Adds a filename variable

This variable is used to add the full filename to the path for either uploading or downloading files

BennettAllen1 committed:
Adds a name for etcd and a name for download to each sub class

BennettAllen1 committed:
Updates all plugins to combine etcd_name and name

These are now the same so a variable is removed

BennettAllen1 committed:
Removes chronos plugin and updates help_info

Removes the unneeded chronos_config_plugin.py
Updates the help_info of all config_plugins to make the argparse output more readable

BennettAllen1 committed:
Updates download name scscf

BennettAllen1 committed:
Changes scscf help info

BennettAllen1 committed:
Updates help_info

BennettAllen1 commented:
@RobDover, not sure if you've reviewed these yet

BennettAllen1 committed:
Removes the outdated upload ifcs_xml's scripts

BennettAllen1 committed:
Updates plugins for coverage

Also remove wrong bgcf plugin
